docker/docker_jit_monitor/src/main.rs

```diff
@@ -184,22 +207,77 @@ fn spawn_runner(config: &RunnerConfig) -> Result<process::Child, SpawnRunnerErro
         .arg(" --jitconfig")
         .arg(encoded_jit_config);
 
-    let runner = cmd
-        .spawn()
-        .map_err(|e| SpawnRunnerError::SpawnDockerError(e))?;
+    let runner = cmd.spawn().map_err(SpawnRunnerError::SpawnDockerError)?;
     Ok(runner)
 }
 
+/// Response of github api runner list
+#[allow(unused)]
+#[derive(Debug, Deserialize)]
+struct ListRunnersResponse {
+    total_count: usize,
+    runners: Vec<GithubRunner>,
+}
+
+/// A github runner from the api
+#[allow(unused)]
+#[derive(Debug, Deserialize)]
+struct GithubRunner {
+    id: u64,
+    name: String,
+    os: String,
+    status: String,
+    busy: bool,
+}
+
+/// Deregisters and kills runners that are offline according to gh api.
+/// This does not kill any docker containers.
+/// Notice that the api endpoint might need a slash
+fn kill_offline_runners(servo_ci_scope: &str) -> Result<(), SpawnRunnerError> {
+    let output = call_github_runner_api(servo_ci_scope, "GET", "", None)?;
+    let runner_response: ListRunnersResponse =
+        serde_json::from_slice(&output.stdout).map_err(SpawnRunnerError::ListRunnersDeserialize)?;
+
+    info!("All runners {runner_response:?}");
+    let filtered_response = runner_response
+        .runners
+        .iter()
+        .filter(|runner| runner.name.contains("dresden-hos"))
```

Comment:
We should probably also limit this to the specific instance, e.g. runner monitor for CI-2, should only clean up jobs from CI-2 (the other CI runner might be running fine for all we know)